### PR TITLE
feat: add vite-plugin-pages-sitemap

### PR DESCRIPTION
**Pull request by jbaubree**

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -80,7 +88,7 @@ export default defineConfig({
     // https://github.com/antfu/vite-plugin-pwa
     VitePWA({
       registerType: 'autoUpdate',
-      includeAssets: ['favicon.svg', 'robots.txt', 'safari-pinned-tab.svg'],
+      includeAssets: ['favicon.svg', 'sitemap.xml', 'robots.txt', 'safari-pinned-tab.svg'],
```
Comment (userquin) [inline vite.config.ts:91]: should be included on the sw precache manifest? this file should be only used by crawlers, do you have some link about this?

File: public/robots.txt [hunk 2]
```diff
@@ -1,2 +1,4 @@
-User-agent: *
+User-agent: *
```
Comment (antfu) [inline public/robots.txt:1]:
```suggestion
User-agent: *
```

**Comment by jbaubree** [comment]
Thanks @antfu i didn't get space error, this is fixed on v1.1.11 https://github.com/jbaubree/vite-plugin-pages-sitemap/commit/e7cd3237fbb291ad94dda21bf1c8774ceb272731

File: vite.config.ts (hunk 1 above)
Comment (jbaubree) [reply vite.config.ts:91]: I'm not sure to understand your question.. i just thinked that sitemap.xml can be read by users, and like robots.txt, this is a needed add.

Comment (userquin) [reply vite.config.ts:91]: `sitemap.xml` is primaly used by `crawlers` to allow search in your site (indexing process for SEO), the browser will not show anything related to the `sitemap.xml` file, you can install some extension to explore it (https://chrome.google.com/webstore/detail/sitemap-explorer/jamphegminpokpnalkjiecfoobdnlmfb).

In fact we should ping crawlers for new deployments: of course anyone can read the `sitemap.xml` and add the sitemap entry to their app. The crawlers will not use the sw and so my question.

you can check this docs:
- https://developers.google.com/search/docs/advanced/sitemaps/overview
- https://developers.google.com/search/docs/advanced/sitemaps/build-sitemap

**Comment by userquin** [comment]
I think we should use it in a custom script to just provide the html files once we finish the build, something similar to `vite-ssg` when regenerating the pwa once the SSG build process finish.

**Comment by userquin** [comment]
we can use `fast-glob` to traverse and find html files on `dist` folder removing the html extension and change `index.html` with the `/` path.

**Comment by jbaubree** [comment]
You are thinking about a vite plugin who works when app is builded? This is an interesting idea

File: README.md [hunk 3]
```diff
@@ -154,6 +155,7 @@ When you use this template, try follow the checklist to update your info properl
 
 - [ ] Change the author name in `LICENSE`
 - [ ] Change the title in `App.vue`
+- [ ] Change the hostname in `vite.config.ts`
```
Comment (jbaubree) [inline README.md:158]: @userquin vite-plugin-pages-sitemap automatically rewrite robots.txt with this hostname

File: vite.config.ts (hunk 1 above)
Comment (jbaubree) [reply vite.config.ts:91]: If the browser do not show anything related to the sitemap, do `robots.txt` have to be removed from VitePWA includeAssets?

Comment (userquin) [reply vite.config.ts:91]: good question

Comment (jbaubree) [reply vite.config.ts]: You are right, i have removed `sitemap.xml` and `robots.txt` in commit ebee2d81f9ad4836b2d13df6c47337e5ebf4c21a

Comment (userquin) [reply vite.config.ts]: Check if offline works, chrome shouln't complain

Comment (jbaubree) [reply vite.config.ts]: No problem in offline mode on my side

**Comment by jbaubree** [comment]
What do you think about gitignore `sitemap.xml` and `robots.txt`? i dont think this files need to be on git, they juste have to be here in dev mode and on build for google robots.

**Review by shorty3613 — APPROVED** [review]

**Pull request merged**